pkg/controller/storageprofile-controller.go

```diff
@@ -0,0 +1,201 @@
+package controller
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/operator"
+	"kubevirt.io/containerized-data-importer/pkg/storagecapabilities"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// StorageProfileReconciler members
+type StorageProfileReconciler struct {
+	client client.Client
+	// use this for getting any resources not in the install namespace or cluster scope
+	uncachedClient client.Client
+	scheme         *runtime.Scheme
+	log            logr.Logger
+}
+
+// Reconcile the reconcile.Reconciler implementation for the StorageProfileReconciler object.
+func (r *StorageProfileReconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithValues("StorageProfile", req.NamespacedName)
+	log.Info("reconciling StorageProfile")
+
+	storageClass := &storagev1.StorageClass{}
+	if err := r.client.Get(context.TODO(), req.NamespacedName, storageClass); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return r.reconcileStorageProfile(storageClass)
+}
+
+func (r *StorageProfileReconciler) reconcileStorageProfile(sc *storagev1.StorageClass) (reconcile.Result, error) {
+	log := r.log.WithValues("StorageProfile", sc.Name)
+	var prevStorageProfile runtime.Object
+
+	storageProfile, err := r.getStorageProfile(sc)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			storageProfile, err = r.createEmptyStorageProfile(sc)
```

Comment:
Either check/use this err, or use _

Reply:
No definitely do
```go
if err != nil {
  return reconcile.Result{}, err
}

Reply:
Thans, editor even gives me the warning, how could I miss that.
```
'storageProfile' may have 'nil' or other unexpected value as its corresponding error variable may be not 'nil' 
```

Reply:
DONE